### ISSUE TITLE
Sharing Embedding meets ERROR!

### ISSUE THREAD

**Issue by BigFishMaster** [OpenNMT/OpenNMT-tf#363]
when I set "share_embeddings=EmbeddingsSharingLevel.ALL" in Transformer model, The trained model can not be correctly infered, e.g. the model will output serveral same tokens for each input query. There may be a bug under this parameter. I also tried share_embeddings=EmbeddingsSharingLevel.NONE, it's ok. 
my env:
gpu: k40
os:redhat
cuda:9.0
opennmt-tf version: v1.20.1 and the current master.

**Comment by guillaumekln** [comment]
Did you make sure to train with a joint vocabulary? If yes, how does the training loss look like with `share_embeddings=EmbeddingsSharingLevel.ALL`?

**Comment by BigFishMaster** [comment]
Yes.My task is Dialogue so I use same vocabulary for input and target. The loss curve seems well but the problem only happens during inference. 

I guess the embedding sharing has something bug. Please help me the fix that. 

The embedding sharing Transformer I used is just from config/models/transformer_share_embeddings.py

**Comment by BigFishMaster** [comment]
@guillaumekln

**Comment by BigFishMaster** [comment]
@guillaumekln

**Comment by guillaumekln** [comment]
Yeah looks like there is a bug. Thanks for testing and reporting, that's helpful.
